### PR TITLE
Handle infinite C_a in AADForest

### LINKED ISSUE #258: Handle `C_a = +np.inf` correctly in AAD
From mathematical point of view, `C_a = +np.inf`  has special meaning that there is no loss for misclassified nominals.
Currently, `C_a = 0` (no loss for misclassified anomalies) is handler correctly automatically.

### PR DESCRIPTION
**Pull request by Eddie-dk1**
Fixes #258.

This handles the C_a=+np.inf AAD edge case without passing non-finite coefficients to the QP solver. In that limit, anomaly loss remains normalized while nominal loss is ignored, matching the issue description.

Added regression coverage for:
- finite weights and scores after fit_known with C_a=np.inf
- equivalence between mixed anomaly/regular labels and anomaly-only labels when nominal loss is ignored

Local checks:
- .venv/bin/python -m pytest tests/test_aadforest.py::test_infinite_c_a_produces_finite_scores tests/test_aadforest.py::test_infinite_c_a_ignores_nominal_loss -q
- .venv/bin/python -m pytest tests/test_aadforest.py -m "not long" -q
- .venv/bin/python -m ruff check src/coniferest/aadforest.py tests/test_aadforest.py
- PY_IGNORE_IMPORTMISMATCH=1 .venv/bin/python -m pytest tests -m "not long" -q: 27 passed, 1 failed due to SSL certificate verification while downloading external DevNet dataset

### CODE REVIEW

**Comment by matwey** [comment]
Hello,

Thanks for the PR.

It is not quite correct way to solve the issue. Your implementation is inconsistent with current behavior. There are three terms in the loss function: 1) anomalies (`C_a`) 2) nominals 3) regularization (`prior_influence`). When I set large (but finite) C_a, it makes both other terms vanish. In your code, `np.inf` makes anomalies compared to regularization.

I suggest deal with the problem differently. At the first step, you introduce new explicit parameter `C_n` for nominals. At the second step, you can do renormalization in AADForest constructor: if any of three parameters is infinite then you replace other to zeroes and if there are more than one infinite parameters, then you raise ValueError.

**Comment by Eddie-dk1** [comment]
Thanks for the review. I updated the PR as suggested:

  - added explicit C_n for nominal loss
  - normalized C_a, C_n, and prior_influence so a single +inf coefficient dominates and the other two become zero
  - raise ValueError when more than one coefficient is infinite
  - kept callable prior_influence support and added regression tests

File: src/coniferest/aadforest.py [hunk 1]
```diff
@@ -12,10 +12,55 @@
...
 
 
+def _validate_coefficient(value, name):
+    if not isinstance(value, Real):
+        raise ValueError(f"{name} is not a real number")
+
+    value = float(value)
+
+    if np.isnan(value) or np.isneginf(value):
+        raise ValueError(f"{name} must be finite or positive infinity")
```
Comment (matwey) [inline src/coniferest/aadforest.py]: You allow coefficients to be negative, but doesn't allow negative infinity? Why? It is inconsistent.

File: src/coniferest/aadforest.py [hunk 2]
```diff
@@ -12,10 +12,55 @@
...
+    infinite_C_n = np.isposinf(C_n)
+
+    if infinite_C_a and infinite_C_n:
+        raise ValueError("At most one of C_a, C_n and prior_influence can be infinite")
+    if infinite_C_a:
+        return 1.0, 0.0, True
+    if infinite_C_n:
+        return 0.0, 1.0, True
+
+    return C_a, C_n, False
```
Comment (matwey) [inline src/coniferest/aadforest.py]: `prior_influence` is floating point, why do you return `True` and `False` here?

File: src/coniferest/aadforest.py [hunk 3]
```diff
@@ -73,12 +118,14 @@ def score_samples(self, x, weights=None):
...
-        q_tau = self._q_tau(scores)
-        known_leafs = self.apply(known_data)
 
         n_anomalies = np.count_nonzero(known_labels == Label.ANOMALY)
         n_nominals = np.count_nonzero(known_labels == Label.REGULAR)
         prior_influence = self.prior_influence(n_anomalies, n_nominals)
+        C_a, C_n, prior_influence = _normalize_loss_coefficients(self.C_a, self.C_n, prior_influence)
+
+        q_tau = self._q_tau(scores, prior_influence)
+        known_leafs = self.apply(known_data)
```
Comment (matwey) [inline src/coniferest/aadforest.py:123]: Why do you do normalization at each `fit_known`? It is enough to do it in constructor, since `C_a`, `C_n` and `prior_influnce` are never changed.

**Review by matwey — CHANGES_REQUESTED** [review]

**Comment by Eddie-dk1** [comment]
I updated the PR according to your comments.

  - coefficients are now consistently required to be non-negative, so finite negative values and -inf are both rejected
  - removed the boolean return from the normalization helper
  - moved the C_a/C_n/prior_influence infinity normalization to the AADForest constructor
  - fit_known now only validates the evaluated callable prior_influence value before passing it to the solver

  I also updated the regression tests for negative coefficients and callable prior_influence validation.

**Comment by hombit** [comment]
@Eddie-dk1 thank you for the effort. I'm closing the PR due to no progress with the implementation. It looks like we are practically talking to an AI agent here, which doesn't have the full context to solve the problem.
